R/data_color.R

```diff
@@ -372,6 +372,55 @@ html_color <- function(colors, alpha = NULL) {
     stop("No values supplied in `colors` should be NA")
   }
 
+  # Ensure that all color names (assumed to be values not preceded
+  # with `#`) are in the set of X11/R color names or CSS color names;
+  # translate certain color names (CSS exclusive names) if necessary
+  if (!all(grepl("^#", colors))) {
+
+    named_colors <- tolower(colors[!grepl("^#", colors)])
```

Comment:
Simplify conditional logic (both 378 and 382) by reusing the variable you'll need anyway.